staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -619,6 +621,248 @@ func RunTestPreconditionalDeleteWithOnlySuggestionPass(ctx context.Context, t *t
+					paginate(ctx, store, &objectCount, &pageCount, limit, resourceVersion, match, field)
+				}
+			} else {
+				paginate(ctx, store, &objectCount, &pageCount, limit, resourceVersion, match, fields.Everything())
+			}
+		}(resourceVersion)
+	}
+	wg.Wait()
+	b.ReportMetric(float64(objectCount.Load())/float64(b.N), "objects/op")
+	b.ReportMetric(float64(pageCount.Load())/float64(b.N), "pages/op")
+}
+
+func paginate(ctx context.Context, store storage.Interface, objectCount, pageCount *atomic.Uint64, limit int64, resourceVersion string, match metav1.ResourceVersionMatch, fields fields.Selector) {
+	listOut := &example.PodList{}
+	err := store.GetList(ctx, "/pods", storage.ListOptions{
+		Recursive:            true,
+		ResourceVersion:      resourceVersion,
+		ResourceVersionMatch: match,
+		Predicate: storage.SelectionPredicate{
+			Field:       fields,
+			GetAttrs:    podAttr,
+			IndexFields: []string{"spec.nodeName"},
+			Label:       labels.Everything(),
+			Limit:       limit,
+		},
+	}, listOut)
+	if err != nil {
+		panic(fmt.Sprintf("Unexpected error %s", err))
+	}
+	pageCount.Add(1)
+	objectCount.Add(uint64(len(listOut.Items)))
+	continueToken := listOut.Continue
+	for continueToken != "" {
+		listOut := &example.PodList{}
+		err := store.GetList(ctx, "/pods", storage.ListOptions{
+			Recursive: true,
+			Predicate: storage.SelectionPredicate{
+				Field:       fields,
+				IndexFields: []string{"spec.nodeName"},
+				GetAttrs:    podAttr,
+				Label:       labels.Everything(),
+				Continue:    continueToken,
+				Limit:       limit,
+			},
+		}, listOut)
+		if err != nil {
+			panic(fmt.Sprintf("Unexpected error %s", err))
+		}
+		continueToken = listOut.Continue
+		pageCount.Add(1)
+		objectCount.Add(uint64(len(listOut.Items)))
+	}
+}
+
+func podAttr(obj runtime.Object) (labels.Set, fields.Set, error) {
+	pod := obj.(*example.Pod)
+	return nil, fields.Set{
+		"spec.nodeName": pod.Spec.NodeName,
+	}, nil
+}
+
+func prepareBenchchmarkData(ctx context.Context, store storage.Interface, namespaceCount, podPerNamespaceCount, nodeCount int) (namespaceNames, nodeNames []string) {
+	nodeNames = make([]string, nodeCount)
+	for i := 0; i < nodeCount; i++ {
+		nodeNames[i] = randomString(100)
+	}
+	namespaceNames = make([]string, nodeCount)
+	out := &example.Pod{}
+	for i := 0; i < namespaceCount; i++ {
+		namespace := randomString(100)
+		namespaceNames[i] = namespace
+		for j := 0; j < podPerNamespaceCount; j++ {
+			name := randomString(100)
+			pod := &example.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name}, Spec: example.PodSpec{NodeName: nodeNames[rand.Intn(nodeCount)]}}
+			err := store.Create(ctx, computePodKey(pod), pod, out, 0)
+			if err != nil {
+				panic(fmt.Sprintf("Unexpected error %s", err))
+			}
+		}
+	}
+	return namespaceNames, nodeNames
+}
+
+const (
+	chars = "abcdefghijklmnopqrstuvwxyz0123456789"
+)
+
+func randomString(l uint) string {
```

Comment:
Instead of creating this function, can you use:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/rand/rand.go#L98

Reply:
Done